src/transformers/cache_utils.py

```diff
@@ -813,7 +813,8 @@ def _quantize(self, tensor, axis):
         if is_optimum_quanto_available():
             from optimum.quanto import quantize_weight
 
-            qtensor = quantize_weight(tensor, self.qtype, axis, self.q_group_size)
+            scale, zeropoint = self.optimizer(tensor, self.qtype, axis, self.q_group_size)
+            qtensor = quantize_weight(tensor, self.qtype, axis, scale, zeropoint, self.q_group_size)
```

Comment:
Why is adding `scale` to `quantize_weight` necessary? What does this fix?

Reply:
optimum-quanto expects `scale` and `shift` to be passed to `quantize_weight`, and the current implementation does not, so quantization fails